fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/mixin/eventslifecycle/MixinEntity.java

```diff
@@ -0,0 +1,28 @@
+package net.fabricmc.fabric.mixin.eventslifecycle;
+
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.entity.EntityTickCallback;
+import net.fabricmc.fabric.impl.event.TickEventInternals;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(Entity.class)
+public abstract class MixinEntity {
+	@Unique
+	private Event<EntityTickCallback<Entity>> tickEvent = TickEventInternals.getOrCreateEntityEvent(this.getClass());
+
+	@Shadow public abstract EntityType<?> getType();
+
+	@Inject(method = "tick", at = @At("TAIL"))
```

Comment:
Notice some entity's tick method does not call super, such as for abstract minecarts.

Reply:
Damn, that is a good remark. I wonder what the best approach is for that very mojang issue.

Reply:
Actually, it seems the best place for the injection would be in the `World#tickEntity(Consumer, Entity)` method. Entities are always ticked through this method in vanilla, however this contract could be broken by other mods.

Reply:
How could it be broken? The world pretty much has to tick everything, yeah?

Reply:
I was thinking a mod that makes entities tick twice a tick may just call `Entity#tick` directly. We can just tell them to use the appropriate world method when that happens, but it should probably be noted somewhere.

Reply:
Ahh, right, time-control mods. Yeah, we should just put it in the docs, since there's not really any way to force all entities to use the tick method.